File: JanperCab/5-JanperCab/ClientApp/src/app/_models/process/ProcessCleaningDto.ts
import { MachineCleaningDto } from '../machine/MachineCleaningDto';
import { ProcessDto } from './ProcessDto';

export class ProcessCleaningDto extends ProcessDto {
  machineId: number;

  machineCleaning: MachineCleaningDto;

  getStatus(): string {
    let status = this.startTime && this.endTime ? 'Cleaned' : 'Cleaning';

    if (this.machineCleaning) {
      status += ` - ${this.machineCleaning.name}`;
    }

    return status;
  }
}

File: JanperCab/3-Application/Dtos/Process/ProcessCleaningDto.cs
using _3_Application.Dtos.Machine;

namespace _3_Application.Dtos.Process
{
    public class ProcessCleaningDto : ProcessDto
    {
        public int? MachineId { get; set; }

        public MachineCleaningDto MachineCleaning { set; get; }
    }
}